examples/interaction_fingerprint.py

```diff
@@ -0,0 +1,284 @@
+    else:
+      search_space['criterion'] = hp.choice('criterion', ["mse", "mae"])
+  elif args['model'] == 'MTR':
+    search_space = {
+        'layer_sizes':
+        hp.choice('layer_sizes', [[3000, 1500], [2000, 1000], [1000, 500]]),
+        'dropout':
+        hp.uniform('dropout', low=0., high=0.5),
+        'lr':
+        hp.uniform('lr', low=1e-4, high=3e-2),
+    }
+  else:
+    raise ValueError('Unexpected model: {}'.format(args['model']))
+
+  return search_space
+
+
+def bayesian_optimization(args):
+  results = []
+  candidate_hypers = init_hyper_search_space(args)
+
+  def objective(hyperparams):
+    configure = deepcopy(args)
+    save_path = init_trial_path(args)
+    val_metrics, test_metrics = main(save_path, configure, hyperparams)
+
+    if args['metric'] in ['roc_auc', 'r2']:
+      # To maximize a non-negative value is equivalent to minimize its opposite number
+      val_metric_to_minimize = -1 * np.mean(val_metrics)
+    else:
+      val_metric_to_minimize = np.mean(val_metrics)
+
+    results.append((save_path, val_metric_to_minimize, val_metrics,
+                    test_metrics))
+
+    return val_metric_to_minimize
+
+  fmin(
+      objective,
+      candidate_hypers,
+      algo=tpe.suggest,
+      max_evals=args['num_trials'])
+  results.sort(key=lambda tup: tup[1])
+  best_trial_path, _, best_val_metrics, best_test_metrics = results[0]
+
+  copyfile(best_trial_path + '/configure.json',
+           args['result_path'] + '/configure.json')
+  copyfile(best_trial_path + '/eval.txt', args['result_path'] + '/eval.txt')
+
+  return best_val_metrics, best_test_metrics
+
+
+if __name__ == '__main__':
+  import argparse
+
+  from utils import mkdir_p
+
+  parser = argparse.ArgumentParser(
+      'Examples for MoleculeNet with interaction fingerprint')
+  parser.add_argument(
+      '-d', '--dataset', choices=['PDBbind'], default='PDBbind', 
+      help='Dataset to use')
+  parser.add_argument(
+      '-m',
+      '--model',
+      choices=['RF', 'MTR'],
+      default='RF',
+      help=
+      'Options include 1) random forest (RF), 2) multitask regressor (MTR) (default: RF)'
+  )
+  parser.add_argument(
+      '-f',
+      '--featurizer',
+      choices=['flat_combined', 'voxel_combined', 'all_combined',
+      'ecfp_ligand', 'ecfp_hashed', 'ecfp', 'splif'],
+      default='flat_combined',
+      help=
+      'Options include 1) flat (ecfp, splif, hbond), 2) voxel (ecfp, splif, salt bridge, charge, hbond, pi stack, cation pi), 3) all (default: flat)'
+  )
+  parser.add_argument(
+      '-p',
+      '--result-path',
+      type=str,
+      default='results',
+      help='Path to save training results (default: results)')
+  parser.add_argument(
+      '-r',
+      '--num-runs',
+      type=int,
+      default=3,
+      help='Number of runs for each hyperparameter configuration (default: 3)')
+  parser.add_argument(
+      '-pa',
+      '--patience',
+      type=int,
+      default=10,
```

Comment:
Why did you use 10 instead? Also the doc remains to be the same.

Reply:
For pdbbind I found that the validation error converges rapidly and the models overfit quickly. It also saves a good amount of time to decrease the patience, because each epoch takes awhile. I'll update the doc with this default value.